### PR TITLE
KFSPTS-30592 Add Vendor Employee Comparison Jobs

### PR DESCRIPTION
**Pull request by cah292**
This PR adds two new KFS batch jobs. One job generates a file containing vendors that might be Cornell employees, then stages it for upload to Workday so that they can determine which vendors indeed represent current or recent Cornell employees. The other job takes the search results file that we receive back from Workday, then generates a simple report that formats the file's rows and computes a few statistics. (At this phase, we're only interested in performing the vendor employee search and viewing its results; any business processing of the results will be handled in future user stories or through manual functional work.)

Some existing OJB utility code has been reworked to simplify the data querying portion, which also required updating one unrelated area that was using the same code. (I also tweaked the Checkstyle suppression rules as a result of that.)

To simplify some of the unit testing portions, I expanded our existing `SpringEnabledMicroTestBase` feature and created a JUnit-5-compatible variant (which is set up as an extension instead of a base test class). The new setup should be more flexible and it allows for using our custom-Spring-context feature in JUnit 5 test classes. I also added some other unit test helper code, such as a JUnit 5 extension for creating/destroying test file directories, and some utility classes for using annotations as test fixtures.

I realize that there's a significant amount of new code introduced by this PR, especially for the unit tests that validate the report generation part. (Also, aside from a few statistics, the new report doesn't add a whole lot more over what the functionals can see by opening Workday's results file via Excel.) If you have a strong opinion about narrowing the scope of this user story to just the generate-search-file part, so that the generate-report part would either be deferred to a follow-up ticket or removed entirely, please let me know.

### CODE REVIEW

**Review by ajd299 — CHANGES_REQUESTED** [review]
While this is a large commit with a large amount of files I do not think that it needs to be split. The code is well designed and easy to understand. Please see my initial feedback

File: src/main/java/edu/cornell/kfs/kew/routeheader/dao/impl/CuDocumentRouteHeaderDAOOjbImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/kew/routeheader/dao/impl/CuDocumentRouteHeaderDAOOjbImpl.java:46]: I agree that the last (or only) statement in the try-with-resources declaration part doesn't need to have a semicolon, though I'm wondering if adding it makes the code easier to understand. Perhaps we could discuss this with the team in one of our meetings tomorrow.

File: src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java:25]: The expressions in the enum's other constants are set up as lambdas since they don't have a straightforward way of being defined as method references (aside from adding methods that may introduce too much clutter). I was trying to keep these as lambdas for consistency, but if you think a method reference would still be better, I could change this accordingly.

File: src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java:29]: If the other item above gets updated to use a method reference instead, then I'll also update this one likewise.

File: src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceImpl.java:64]: Good catch, I'll fix the typo.

File: src/main/java/edu/cornell/kfs/vnd/businessobject/VendorEmployeeComparisonResult.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/vnd/businessobject/VendorEmployeeComparisonResult.java:48]: These result objects will ultimately be generated from a .csv file returned by Workday, so I had set them up as `Boolean` properties to help clarify whether a null value was sent back to us. However, since we're expecting Workday to return non-null values in this field, I'll look into updating this to use primitive booleans instead, and then have the conversion either translate nulls to false or treat nulls as errors.

File: src/main/resources/edu/cornell/kfs/vnd/cu-vnd-resources.properties
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/vnd/cu-vnd-resources.properties:22]: These properties are being used by the new `VendorEmployeeComparisonReportServiceImpl` class, along with its related unit tests.

File: src/test/java/edu/cornell/kfs/sys/util/CreateTestDirectoriesExtension.java
Comment (cah292) [reply src/test/java/edu/cornell/kfs/sys/util/CreateTestDirectoriesExtension.java:37]: I agree that this `throws` is not needed; I'll remove it.

File: src/test/java/edu/cornell/kfs/sys/util/TestSpringContextExtension.java
Comment (cah292) [reply src/test/java/edu/cornell/kfs/sys/util/TestSpringContextExtension.java:96]: I'll remove the unneeded `throws`.

File: src/test/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceOutboundFileTest.java
Comment (cah292) [reply src/test/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceOutboundFileTest.java:263]: Depending on what's decided in our upcoming team discussion, I'll remove the semicolon if necessary.

File: src/test/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceResultFileTest.java
Comment (cah292) [reply src/test/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceResultFileTest.java:463]: I think most or all of the other areas that are still lambdas have been kept that way to make things easier to understand, but for this particular line, I agree that it makes sense to use a method reference instead. I'll update it accordingly.

File: src/test/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceResultFileTest.java
Comment (cah292) [reply src/test/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonServiceResultFileTest.java:487]: Will remove the semicolon if necessary, depending on the team discussion.

File: src/test/java/edu/cornell/kfs/vnd/batch/service/impl/fixture/VendorComparisonResultRowFixture.java
Comment (cah292) [reply src/test/java/edu/cornell/kfs/vnd/batch/service/impl/fixture/VendorComparisonResultRowFixture.java:67]: The semicolon's not strictly needed here since the enum doesn't include extra methods or fields. However, due to the other annotation-based data that's being included along with the constants, I would prefer to keep this particular semicolon in place.

File: src/main/java/edu/cornell/kfs/vnd/businessobject/VendorEmployeeComparisonResult.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/vnd/businessobject/VendorEmployeeComparisonResult.java:48]: Actually, upon further review, I remembered that I had set up the reporting so that it would warn about missing data in this field, so it needs to remain `Boolean` to simplify that reporting.

File: src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java:25]: It looks like simplifying this one to a method reference is feasible, so I'll proceed with that.

File: src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/vnd/batch/service/impl/VendorEmployeeComparisonReportServiceImpl.java:29]: Updating.

**Comment by cah292** [comment]
I've committed the first round of revisions, but I'm waiting for a developer discussion tomorrow before wrapping up the remaining portions (such as whether to remove some of the try-with-resources semicolons).

File: src/main/java/edu/cornell/kfs/kew/routeheader/dao/impl/CuDocumentRouteHeaderDAOOjbImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/kew/routeheader/dao/impl/CuDocumentRouteHeaderDAOOjbImpl.java:46]: Upon further consideration, it looks like KualiCo is excluding the semicolon from the last (or only) statements in its try-with-resources declarations, so I'll update the relevant blocks in this PR for consistency.

**Comment by cah292** [comment]
I think I've finished adjusting the remaining areas that needed modifications. If there's a section that you still think needs further changes, please let me know.

Also, if you haven't already, please review the draft OpDocs attached to the review sub-task. I still need to get Prod Control's input on them so that I can fill in the Procedure IDs, but overall they should be mostly complete now.

This is ready for review again.

**Comment by ajd299** [comment]
Okay, code looks good, I will merge when I OpDocs are ready

**Comment by cah292** [comment]
The latest copies of the OpDocs have been attached to the main user story. I have one pending question to the functionals concerning the information in those docs, but other than that those two latest copies should have all the needed information in place.

**Review by ajd299 — APPROVED** [review]
I approve, can you confirm this is ready to be merged to develop branch

**Comment by cah292** [comment]
Yes, if you approve of the changes, then this should be ready to merge now. The current remaining OpDoc items that are being sorted out don't need to hold up the code changes. Thanks!

**Pull request merged**
